### PR TITLE
[C-2496, C-2497] Refactor Playlist Library UI

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Large overhaul to playlist-library ui to improve readability, performance, and useability.

- Defined core components used across left nav: `LeftNavItem` and `LeftNavDroppable`
- Implemented core nav items from this: `PlaylistNavItem` `PlaylistFolderNavItem` `SmartCollectionNavItem` `NftAudioPlaylistNavItem`

- Refactored large complex components like `PlaylistLibrary` into smaller chunks, maximizing reuse and perf
- Moved some of the complex library reoder operations to sagas, since they needed full access to the library, which would not be performant defining those interactions in each nav-item.

- Improved drag-drop code by moving away from event listeners, and using react onDrop props
- Added `forward` option to prevent drag/drop components adding extra divs around links

- Fixed bugs related to playlist-library updates: the merge function in the merge customizer was merging arrays, resulting in duplicate values.

### TODO: 
- a11y pass: we should be using ul/li for these links
- Drag-drop below folder needs to be improved (current issues as well)
- Further droppable optimizations using renderProp / cloneElement

### Dragons

Lots of changes, so need to run this feature through it's paces. will be working in this area for next few prs anyway, so will be battle testing it.

### CODE REVIEW

File: packages/web/src/components/dragndrop/Draggable.tsx [hunk 1]
```diff
@@ -40,11 +40,11 @@ export const Draggable = (props: DraggableProps) => {
     children,
     ...otherProps // passed to child
   } = props
-  const draggableRef = useRef<HTMLDivElement | null>(null)
   const dispatch = useDispatch()
 
   const handleDragStart = useCallback(
-    (e: any) => {
+    (e: DragEvent) => {
```
Comment (Kyle-Shanks) [inline packages/web/src/components/dragndrop/Draggable.tsx:46]: nice

File: packages/web/src/components/nav/desktop/PlaylistLibrary/PlaylistUpdateDot.module.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.root {
+  /* margin left = -(6px width + 4px margin-right on update dot element) so that list items stay aligned */
+  margin-left: -10px;
```
Comment (Kyle-Shanks) [inline packages/web/src/components/nav/desktop/PlaylistLibrary/PlaylistUpdateDot.module.css:3]: negative margins always feel off to me for some reason, dont want to shake up the structure that's working though

**Review by Kyle-Shanks — APPROVED** [review]
big, Huge, GIGANTIC!

File: packages/web/src/components/nav/desktop/PlaylistLibrary/PlaylistUpdateDot.module.css (hunk 2 above)
Comment (dylanjeffers) [reply packages/web/src/components/nav/desktop/PlaylistLibrary/PlaylistUpdateDot.module.css:3]: yeah agreed... im down to do a css pass with you at some point with all this if youre game, pretty much all that is copied from previous setup

**Pull request merged**